http/src/main/java/com/github/awsjavakit/http/OAuth2HttpClient.java

```diff
@@ -0,0 +1,95 @@
+package com.github.awsjavakit.http;
+
+import static com.gtihub.awsjavakit.attempt.Try.attempt;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.awsjavakit.misc.paths.UriWrapper;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandler;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Objects;
+
+public class OAuth2HttpClient {
```

Comment:
Most of this code is copy-paste from nva-commons. However, I have created a new interface OAuthCredentialsProviders, so that the logic for getting the credentials is not included here.